iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/IcebergTableOpsManager.java

```diff
@@ -35,16 +43,41 @@ public class IcebergTableOpsManager implements AutoCloseable {
   private static final ImmutableMap<String, String> ICEBERG_TABLE_OPS_PROVIDER_NAMES =
       ImmutableMap.of(
           ConfigBasedIcebergTableOpsProvider.CONFIG_BASE_ICEBERG_TABLE_OPS_PROVIDER_NAME,
-          ConfigBasedIcebergTableOpsProvider.class.getCanonicalName());
+          ConfigBasedIcebergTableOpsProvider.class.getCanonicalName(),
+          GravitinoBasedIcebergTableOpsProvider.GRAVITINO_BASE_ICEBERG_TABLE_OPS_PROVIDER_NAME,
+          GravitinoBasedIcebergTableOpsProvider.class.getCanonicalName());
 
   private final Cache<String, IcebergTableOps> icebergTableOpsCache;
 
   private final IcebergTableOpsProvider provider;
 
   public IcebergTableOpsManager(Map<String, String> properties) {
-    this.icebergTableOpsCache = Caffeine.newBuilder().build();
     this.provider = createProvider(properties);
     this.provider.initialize(properties);
+    this.icebergTableOpsCache =
+        Caffeine.newBuilder()
+            .expireAfterWrite(
+                (new IcebergConfig(properties))
+                    .get(IcebergConfig.ICEBERG_REST_CATALOG_CACHE_EVICTION_INTERVAL),
+                TimeUnit.MILLISECONDS)
+            .removalListener(
+                (k, v, c) -> {
+                  LOG.info("Remove IcebergTableOps cache {}.", k);
+                  try {
```

Comment:
We could make GravitinoCatalogProvider implement `Closable`, and close it in remove listener
```java
if (v instanceOf Closeable) {
  ((Closable)v).close()
}
```

Reply:
I modified GravitinoCatalogProvider to maintain a GravitinoAdminClient and ensure it closes when IcebergTableOpsManager closes.